File: my_cpp_pkg/src/add_two_ints_client_noop.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("add_two_ints_client_noop");

    auto client = node->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
    while(!client->wait_for_service(std::chrono::seconds(1)))
    {
        RCLCPP_WARN(node->get_logger(),"Esperando o server iniciar!");
    }

    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = 4;
    request->b =18;

    auto future = client->async_send_request(request);
    if(rclcpp::spin_until_future_complete(node,future) == rclcpp::FutureReturnCode::SUCCESS)
    {
        RCLCPP_INFO(node->get_logger(),"%ld + %ld =%ld", request->a, request->b, future.get()->sum);
    }
    else
    {
        RCLCPP_ERROR(node->get_logger(), "Erro ao chamar o serviço!");
    }
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/add_two_ints_client_noop.py
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts
 
 
def main(args=None):
    rclpy.init(args=args)
    node = Node("add_two_ints_no_opp")
    client = node.create_client(AddTwoInts, "add_two_ints")

    while not client.wait_for_service(1.0): #Aguarda até que o server esteja pronto  
        node.get_logger().warn("Esperando pelo server Add Two Ints") #Envia o warning a uma taxa de 1hz

    request = AddTwoInts.Request()
    request.a = 4
    request.b = 5

    future = client.call_async(request)
    rclpy.spin_until_future_complete(node, future)

    try:
        response = future.result()
        node.get_logger().info(str(request.a) + " + " + str(request.b) + 
                                                " = " + str(response.sum))
    except Exception as erro:
        node.get_logger().error("Chamada de servico falhou!" % (erro,))

    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()